packages/AlertDialog/index.tsx

```diff
@@ -0,0 +1,161 @@
+import React, {useState} from 'react';
+import Modal, {ModalProps} from 'react-native-modalbox';
+import styled from '@emotion/native';
+
+import {StyleProp, ViewStyle, TextStyle} from 'react-native';
+
+const ModalButton = styled.TouchableOpacity<ButtonTypeProps>`
+  width: 100%;
+  height: 40px;
+  border: 1px solid black;
+  border-radius: 20px;
+  justify-content: center;
+  align-items: center;
+  background-color: ${(props) =>
+    props.isAdditional ? 'transparent' : 'black'};
+  margin-bottom: ${(props) => (props.isAdditional ? '16px' : '0px')}; ;
+`;
+
+const ButtonText = styled.Text<ButtonTypeProps>`
+  font-size: 14px;
+  font-weight: bold;
+  color: ${(props) => (props.isAdditional ? 'black' : 'white')}; ;
+`;
+
+const ModalInput = styled.TextInput`
+  height: 40px;
+  margin-bottom: 16px;
+  padding: 4px;
+  font-size: 14px;
+  border: 1px solid black;
+`;
+
+const Title = styled.Text`
+  font-size: 17px;
+  line-height: 20px;
+  text-align: center;
+  margin-bottom: 16px;
+`;
+
+const Content = styled.Text`
+  font-size: 16px;
+  line-height: 20px;
+  margin-bottom: 16px;
+`;
+
+type Styles = {
+  modal?: ModalProps['style'];
+  button?: StyleProp<ViewStyle>;
+  buttonText?: StyleProp<TextStyle>;
+};
+
+interface ButtonTypeProps {
+  isAdditional?: boolean;
+}
+
+interface Props {
+  isOpen?: boolean;
+  type?: 'alert' | 'confirm' | 'prompt';
+  title?: string;
+  content?: string;
+  styles?: Styles;
+  backdropOpacity?: number;
+  backdropPressToClose?: boolean;
+  onPress: (result: boolean | string | null) => void;
+}
+
+const AlertDialog = React.forwardRef(
+  (
+    {
+      isOpen = false,
+      type = 'alert',
+      title,
+      content,
+      styles = {},
+      backdropOpacity = 0.5,
+      backdropPressToClose = true,
+      onPress,
+    }: Props,
+    ref: React.RefObject<Modal>,
+  ) => {
+    const [input, setInput] = useState('');
+
+    const renderAlert = (
+      <ModalButton style={styles?.button} onPress={() => onPress(true)}>
```

Comment:
I try not to repeat same word. In here, `style` is repeating.
@hyochan [mentioned](https://github.com/dooboolab/dooboo-ui/pull/105#discussion_r704329141) it recently. 

(I think I read about it in [`refactoring` ](https://martinfowler.com/books/refactoring.html)too but I can't remember 😢 )

So I suggest you to restructure `styles`. ([example](https://github.com/dooboolab/dooboo-ui/blob/b69c16dea2a231705f66d8f8c5763952499d58f0/main/SwitchToggle.tsx#L52-L76))